source/appModules/windowsinternal_composableshell_experiences_textinput_inputapp.py

```diff
@@ -11,19 +11,109 @@
 
 import appModuleHandler
 import api
+import eventHandler
 import speech
 import braille
 import ui
 import config
 import winVersion
+import controlTypes
 from NVDAObjects.UIA import UIA
+from NVDAObjects.behaviors import CandidateItem as CandidateItemBehavior
+
+
+class ImeCandidateUI(UIA):
+	"""
+	The UIAutomation-based IME candidate UI (such as for  the modern Chinese Microsoft Quick input).
+	This class ensures NVDA is notified of the first selected item when the UI is shown.
+	"""
+
+	def event_show(self):
+		# The IME candidate UI is shown.
+		# Report the current candidates page and the currently selected item.
+		# Sometimes UIA does not fire an elementSelected event when it is first opened,
+		# Therefore we must fake it here.
+		if (

```

Comment:
```suggestion
		if (self.UIAAutomationId == "IME_Prediction_Window"):
			candidateItem = self.firstChild
			eventHandler.queueEvent("UIA_elementSelected", candidateItem)
		elif (

```
New Microsoft Pinyin IME in Windows 10 20H1 emits similar events with New Microsoft Quick. New Microsoft Pinyin IME input method can also benefit from this pr with some modifications.